pintc/tests/optimizations/const_folding.pnt

```diff
@@ -89,8 +89,7 @@ predicate Test {
 //     constraint (::d == true);
 //     constraint (::e == -23);
 //     constraint (::f == (::d ? false : true));
-//     constraint (true || (0 == ::y));
-//     constraint (false || (((::y == 1) || (::y == 4)) || (::y == 7)));
+//     constraint (((::y == 1) || (::y == 4)) || (::y == 7));
```

Comment:
Unfortunately the new logic affects the `const_folding` test as well. I couldn't think of a way for it not to. When there is an `if statement` with a `const` condition, it's guaranteed to optimize away the `constraint true` and `constraint false` of the branches

Reply:
Yeah this is fine :)